src/source-napatech.c

```diff
@@ -125,35 +129,65 @@ SC_ATOMIC_DECLARE(uint64_t, flow_callback_tcp_pkts);
 SC_ATOMIC_DECLARE(uint64_t, flow_callback_unhandled_pkts);
 
 /**
- * \brief Register the Napatech  receiver (reader) module.
+ * \brief Initialize the Napatech receiver (reader) module for globals.
  */
-void TmModuleNapatechStreamRegister(void)
+static TmEcode NapatechStreamInit(void)
 {
-    tmm_modules[TMM_RECEIVENAPATECH].name = "NapatechStream";
-    tmm_modules[TMM_RECEIVENAPATECH].ThreadInit = NapatechStreamThreadInit;
-    tmm_modules[TMM_RECEIVENAPATECH].Func = NULL;
-    tmm_modules[TMM_RECEIVENAPATECH].PktAcqLoop = NapatechPacketLoop;
-    tmm_modules[TMM_RECEIVENAPATECH].PktAcqBreakLoop = NULL;
-    tmm_modules[TMM_RECEIVENAPATECH].ThreadExitPrintStats = NapatechStreamThreadExitStats;
-    tmm_modules[TMM_RECEIVENAPATECH].ThreadDeinit = NapatechStreamThreadDeinit;
-    tmm_modules[TMM_RECEIVENAPATECH].cap_flags = SC_CAP_NET_RAW;
-    tmm_modules[TMM_RECEIVENAPATECH].flags = TM_FLAG_RECEIVE_TM;
+    int i;
 
     SC_ATOMIC_INIT(total_packets);
     SC_ATOMIC_INIT(total_drops);
     SC_ATOMIC_INIT(total_tallied);
     SC_ATOMIC_INIT(stream_count);
 
-    SC_ATOMIC_INIT(numa0_count);
-    SC_ATOMIC_INIT(numa1_count);
-    SC_ATOMIC_INIT(numa2_count);
-    SC_ATOMIC_INIT(numa3_count);
+    numa_detect = SCMalloc(sizeof(*numa_detect) * (numa_max_node() + 1));
+    if (numa_detect == NULL) {
+        FatalError("Failed to allocate memory for numa detection array: %s",
```

Comment:
`FatalError` terminates suricata so the `return` following this call isn't neede.

Reply:
Thanks fixed.